output/filewriter.go

```diff
@@ -352,17 +333,65 @@ func newWorker(cfg *FileWriterConfig, tmpl *template.Template, replFieldValue st
 				if _, err := curw.Write(line); err != nil {
 					log.WithError(err).Error("FileWriter worker error writing to file")
 				}
-				const linesep = "\n"
-				if _, err := curw.Write([]byte(linesep)); err != nil {
+
+				const linesep = '\n'
+				if _, err := curw.Write([]byte{linesep}); err != nil {
 					log.WithError(err).Error("FileWriter worker error writing to file")
 				}
+
+				nwritten := bytesWritten()
+				if cfg.RotateSize != 0 && nwritten > int64(cfg.RotateSize) {
+					// Max size reached, we can rotate and reset the 'interval timer'.
+					rotate()
+					restartTicker()
+				}
 			}
 		}
 	}()
 
 	return fw, nil
 }
 
+func (fw *fileWorker) newFile(path string) (io.WriteCloser, func() int64, error) {
+	f, err := os.Create(path)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	bufw := bufio.NewWriterSize(f, fileWorkerChunkBuffer)
+	countw := newCountingWriter(bufw)
```

Comment:
If I understand correctly the code, here we are counting the not compressed bytes, so the final file size could have a lower size than the one selected by the user. This could be the intended behavior, but maybe we could make it clear on the documentation.

Reply:
We're counting the compressed bytes here (i.e the final size of the compressed file). 
The `countingWriter` wraps the `bufio.Writer`, which itselfs wraps (and buffers) the final file. The bytes written into `countw` comes from the zip writer.
The test confirms that -> https://github.com/AdRoll/baker/blob/8271279a50346a20bf2ec98d0ee6f4838983f728/output/filewriter_test.go#L604-L614

Let me know if that's clear or if there's some misunderstanding.

Reply:
You are right I misread the code. Thanks for the clarification.